### PR TITLE
(PUP-8464) update puppet help to process its arguments

### PR DESCRIPTION
**Pull request by ericldelaney**
Added better argument handling
Updated a some error messages to be more user friendly
Fixed some I18n messages

### CODE REVIEW

**Comment by ericldelaney** [comment]
jenkins please test this

File: spec/unit/face/help_spec.rb [hunk 1]
```diff
@@ -122,13 +125,6 @@
           expect(Puppet::Face[name, :current]).to have_a_summary
         end
       end
-
-      it "returns an 'unavailable' summary if the face application raises a LoadError" do
```
Comment (jhelwig) [inline spec/unit/face/help_spec.rb]: There's a version of this test for legacy applications up a little higher in the file. Are face applications no longer supposed to do this?

Comment (ericldelaney) [reply spec/unit/face/help_spec.rb]: @jhelwig  Ok, I added context's around the legacy tests at the at new line 190 and we cover subcommand and action failures at line 50 and 69 respectively.
I also converted the double quotes to singles while I was at it.

**Comment by joshcooper** [comment]
Since `puppet help agent` prints help about agent, I would expect `puppet help help` to print help about the help application. This also follows from the global help:

```
See 'puppet help <subcommand>' for help on a specific subcommand.
```

However, when `puppet help help` is called the `when_invoked` block is called with `args`:

```
(byebug) args
[{}]
```

But when it's called with `puppet help config`, the `args` are different:

```
(byebug) args
["config", {}]
```

So it seems like something is "eating" the subcommand name. And it looks like it's because `help help` is the default action, which triggers https://github.com/puppetlabs/puppet/blob/5.4.0/lib/puppet/application/face_base.rb#L178-L187.

**Comment by ericldelaney** [comment]
@joshcooper Yep, face_base eats one of the command line arguments when its the default case. I had noticed that but I was afraid to fix it and break face apps that others have written that expect the behavior.

**Pull request merged**